### PR TITLE
Add Transcription script for all ASR models 

### PR DESCRIPTION
**Pull request by titu1994**
# Changelog
- Add single script to allow transcription of all ASR models
- Add subclass walk for finding all Pretrained models of ASRModel. 
- Add dataclasses for RNNT Decoding strategy

### CODE REVIEW

File: examples/asr/transcribe_speech.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        # restore model from .nemo file path
+        model_cfg = ASRModel.restore_from(restore_path=cfg.model_path, return_config=True)
+        classpath = model_cfg.target  # original class path
+        imported_class = model_utils.import_class_by_path(classpath)  # type: ASRModel
+        logging.info(f"Restoring model : {imported_class.__name__}")
+
+        asr_model = imported_class.restore_from(restore_path=cfg.model_path, map_location=device)  # type: ASRModel
+    else:
+        # restore model by name
+        asr_model = ASRModel.from_pretrained(model_name=cfg.pretrained_name, map_location=device)  # type: ASRModel
```
Comment (blisc) [inline examples/asr/transcribe_speech.py:83]: Can you add an unittest that checks if this works?

Comment (titu1994) [reply examples/asr/transcribe_speech.py:89]: Added, trying to get Jenkins to actually run these tests

Comment (titu1994) [reply examples/asr/transcribe_speech.py:89]: Unittests dont have access to examples/asr/* so had to put the test in Jenkins instead

Comment (titu1994) [reply examples/asr/transcribe_speech.py:89]: Tests now run on this branch - but I had to explicitly set the branch name. Needs to be reverted to main when merged back. @okuchaiev

**Review by blisc — APPROVED** [review]

**Pull request merged**